_write.js

```diff
@@ -57,6 +51,16 @@ module.exports = function _write(httpMethod, options, callback) {
       catch (e) {
         err = e
       }
+
+      if (!ok) {
+        err = Error(httpMethod + ' failed with: ' + res.statusCode)
+        err.raw = res
+        err.body = result
+        callback(err)
+        res.resume()
+        return
```

Comment:
feel it would be more readable to lose the return statement and wrap the callback beneath in an `else` statement

Reply:
Yep, works for me.